README.md

```diff
@@ -109,11 +78,70 @@ s3://mybucket/aws-instance-scheduler/v1.4.0/instance-scheduler.template
 s3://mybucket/aws-instance-scheduler/v1.4.0/instance-scheduler-remote.template
 ```
 
-## Deploy
+### Deploy
 
 See the [AWS Instance Scheduler Implementation Guide](https://s3.amazonaws.com/solutions-reference/aws-instance-scheduler/latest/instance-scheduler.pdf) for deployment instructions, using the link to the instance-scheduler.template from your bucket, rather than the one for AWS Solutions. Ex. https://mybucket.s3.amazonaws.com/aws-instance-scheduler/v1.4.0.mybuild/instance-scheduler.template
 
-## CDK Documentation
+
+# Testing the Solution
+
+## Running Tests Locally
+to test the solution you will need to install tox
+
+```pip install tox```
+
+Then from the root directory of the solution
+```
+//run all unit tests
+tox
+
+//test just the lambda code
+tox -e lambda
+
+//test just the cdk code
+tox -e cdk
+```
+
+## Automated Testing Pipeline
```

Comment:
Questions unanswered by this section
- what does this deploy into my account (a pipeline and the solution, maybe need an arch diagram)
- how do I interpret the results for this
- how does this pipeline get kicked off (by push events)

Reply:
- currently it only deploys the pipeline, deployment testing wont work until we get the solution cdk-deployable
- I would think that's self-explanatory by just clicking on the deployed pipeline (green good, red bad)?
- it will run immediately upon deploy and then upon every push. I implied this by saying "AWS Instance Scheduler comes with an automated testing pipeline that can be used to automatically test any changes you develop for the solution on your own development fork" but if you think it needs to be more explicit I can change that

Reply:
I'm not asking you, I'm saying that I would probably have these questions if I navigate to this repo and read this section. Your readme should address these questions. Sorry if that was unclear.

I disagree that the pipeline results are self-explanatory. Say I've followed these instructions. What got deployed? I don't know. Where is the pipeline? How do I find it? I may not have used codepipeline before. I may not have used cdk deploy before and may not know that it just creates a cloudformation stack. What about failures that are the result of setup not being done correctly? Failures in the source stage are likely because your params or permissions are wrong.

What if I already have the solution deployed in my account? You can deploy instance scheduler multiple times, so there shouldn't be any issues - right?

Also, this is kind of out of scope, but I might want to know what kind of role permissions do I need in order to deploy this pipeline?

Reply:
updated info on how to view the pipeline and more explicitly said that the pipeline runs on all push events to a specified branch. Extra info on solution deploy tests will need to be added after we get that feature working